js.go

```diff
@@ -2766,11 +2756,214 @@ func (sub *Subscription) Fetch(batch int, opts ...PullOpt) ([]*Msg, error) {
 	}
 	// If there is at least a message added to msgs, then need to return OK and no error
 	if err != nil && len(msgs) == 0 {
-		return nil, checkCtxErr(err)
+		return nil, o.checkCtxErr(err)
 	}
 	return msgs, nil
 }
 
+type FetchResult interface {
+	Messages() <-chan *Msg
+	Error() error
+}
+
+type fetchResult struct {
+	msgs chan *Msg
+	err  error
+}
+
+func (fr *fetchResult) Messages() <-chan *Msg {
+	return fr.msgs
+}
+
+func (fr *fetchResult) Error() error {
+	return fr.err
+}
+
+// FetchChan pulls a batch of messages from a stream for a pull consumer.
+// Unlike [Subscription.Fetch], it is non blocking and returns [FetchResult],
+// allowing to retrieve incoming messages from a channel.
+// The returned channel is always close - it is safe to iterate over it using range.
+//
+// To avoid using default JetStream timeout as fetch expiry time, use [nats.MaxWait]
+// or [nats.Context] (with deadline set).
+//
+// This method will not return error in case of pull request expiry (even if there are no messages).
+// Any other error encountered when receiving messages will cause FetchChan to stop receiving new messages.
+func (sub *Subscription) FetchChan(batch int, opts ...PullOpt) (FetchResult, error) {
```

Comment:
How about `FetchBatch` and then the interface is a `MessageBatch`? It would be then like `FetchBatch pulls a batch of messages... and returns a MessageBatch`.

Reply:
I like that 👍

Reply:
Done

Reply:
nice!